packages/vanilla/rollup.config.mjs

```diff
@@ -1,17 +1,36 @@
 import { babel } from '@rollup/plugin-babel'
-import ts from 'rollup-plugin-typescript2'
 import pkg from './package.json' assert { type: 'json' }
+import typescript from '@rollup/plugin-typescript'
 
-const config = {
-  input: 'src/index.ts',
-  output: [
-    { file: pkg.main, format: 'cjs' },
-    { file: pkg.module, format: 'esm' },
-  ],
-  plugins: [
-    ts({ tsconfigOverride: { exclude: ['**/*-test.ts'] } }),
-    babel({ babelHelpers: 'inline' }),
-  ],
-}
+const config = [
+  // config for CommonJS
+  {
+    input: 'src/index.ts',
+    output: { file: pkg.main, format: 'cjs', generatedCode: 'es5' },
```

Comment:
For the CommonJS export, we assume the worst-case and generate ES5 code. If you take a look in `./dist` after `pnpm build`, you'll see polyfills and arrow functions converted to "regular" functions.